lib/Service/Note.php

```diff
@@ -105,6 +105,9 @@ public function getData(array $exclude = []) : array {
 		if (!in_array('readonly', $exclude)) {
 			$data['readonly'] = $this->getReadOnly();
 		}
+		$data['internalPath'] = $this->file->getInternalPath();
+		$data['shareTypes'] = $this->noteUtil->getShareTypes($this->file);
```

Comment:
Should allow to achieve the same without triggering heavy loading of all the share entries from the db:
```suggestion
		$data['shareTypes'] = $this->file->isShared();
```